src/GitVersionCore.Tests/IntegrationTests/DevelopScenarios.cs

```diff
@@ -284,5 +284,50 @@ public void PreviousPreReleaseTagShouldBeRespectedWhenCountingCommits()
 
             fixture.AssertFullSemver("1.0.0-alpha.5");
         }
+
+        [Test]
+        public void WhenPreventIncrementOfMergedBranchVersionIsSetToTrueCommitsSinceVersionSourceShouldNotGoDownWhenMergingReleaseToDevelop()
+        {
+            // Simulate the configuration used in production.
+            var config = new Config
+            {
+                VersioningMode = VersioningMode.ContinuousDeployment,
+                Branches = new Dictionary<string, BranchConfig>
+                {
+                    { "develop", new BranchConfig { PreventIncrementOfMergedBranchVersion = true } },
+                }
+            };
+
+            using var fixture = new EmptyRepositoryFixture();
+            const string ReleaseBranch = "release/1.1.0";
+            fixture.MakeACommit();
+            fixture.BranchTo("develop");
+            fixture.MakeATaggedCommit("1.0.0");
+            fixture.Repository.MakeCommits(10);
+
+            fixture.AssertFullSemver("1.1.0-alpha.10", config);
+
+            // Create a release branch and make some commits
+            fixture.BranchTo(ReleaseBranch);
+            fixture.Repository.MakeCommits(3);
+
+            // Simulate a GitFlow release finish.
+            fixture.Checkout("master");
+            fixture.MergeNoFF(ReleaseBranch);
+            fixture.ApplyTag("v1.1.0");
+            fixture.Checkout("develop");
+            // Simulate some work done on develop while the release branch was open.
+            fixture.Repository.MakeCommits(10);
+            fixture.MergeNoFF(ReleaseBranch);
+
+            // Version numbers will still be correct when the release branch is around.
+            fixture.AssertFullSemver("1.2.0-alpha.14");
+            fixture.AssertFullSemver("1.2.0-alpha.14", config);
+
+            // Removing the release branch and calculating the version with the config exhibits the problem.
+            fixture.Repository.Branches.Remove(ReleaseBranch);
```

Comment:
I agree that removal of the release branch should not change the version number. Can you add an assert of the version source as well, just to ensure that it doesn't change when the release branch is removed? If the version source is the same, the version number should also stay the same.

Reply:
@asbjornu I have extracted the commit hashes before and after the deletion of the release branch and asserted equality. I hope that is what you meant.

Reply:
Ah yes, I see now the comparison on line 332. Sorry.